doc/explanation/api/reactivity.md

```diff
@@ -0,0 +1,177 @@
+
+:::{note}
+Panel has gone through a significant evolution over time and many of the concepts discussed here really only fully crystallized when Param 2.0 was released.
+:::
+
+## Concepts
+
+The core concept a large part of the discussion will focus on is the difference between reactive and event driven approaches for writing UIs. Panel has placed strong emphasis on innovating to provide novel **reactive** approaches to writing applications. Popular reactive frameworks include spreadsheets like Excel and frontend UI frameworks like React, each of which has a different approach to reactivity, but two core principles are conserved:
+
+1. They are **data-driven**, i.e. the UI updates automatically in response to changes in the data model. In Excel the data model is represented by value of a cell, while React has use state hooks. In Panel the data model is defined by Parameters, i.e. components and user classes define some parameters which drive changes in the UI. In all these frameworks this is achieved by what we will call **data binding**, how precisely this works will be the primary focus of this section.
+
+2. They are **declarative**, i.e. the user, defines what should be rendered and the reactive framework figures out how to efficiently make the required updates. In Excel this is simple, it simply has to re-evaluate any formulas that depend on the changed inputs and then update the cell's output. In React this can be significantly more involved, requiring diffing of the document model and then figuring out the most efficient updates to reflect the latest changes. In Panel this works either by defining a function who's output is diffed or by binding references to a declarative component (if you don't know what that means yet, don't worry, that's what this guide is for).
+
+The alternative to the reactive approach is an event-driven approach, involving callbacks that are triggered by user interactions or other events. This approach usually relies on **imperative** rather than **declarative** programming, i.e. you define step-by-step what should happen when a particular action is triggered. This is usually significantly more cumbersome, especially as an application grows in size and complexity. Even in a relatively simple application where we gather user inputs from multiple widgets and then perform some update we quickly run into problems. Let us take a simple example of an application that has three widgets that gather user input. In a callback based approach we have to register a callback with each of the three widgets and then perform some action as a result:
```

Comment:
```suggestion
The alternative to the reactive approach is an **event-driven** approach, involving callbacks that are triggered by user interactions or other events. This approach usually relies on **imperative** rather than **declarative** programming, i.e. you define step-by-step what should happen when a particular action is triggered. This is usually significantly more cumbersome, especially as an application grows in size and complexity. Even in a relatively simple application where we gather user inputs from multiple widgets and then perform some update we quickly run into problems. Let us take a simple example of an application that has three widgets that gather user input. In a callback based approach we have to register a callback with each of the three widgets and then perform some action as a result:
```